libarchive/archive_read_support_filter_uu.c

```diff
@@ -40,14 +40,14 @@
 #include "archive_private.h"
 #include "archive_read_private.h"
 
-/* Maximum lookahead during bid phase */
-#define UUENCODE_BID_MAX_READ 128*1024 /* in bytes */
+/* Maximum line length supported */
+#define UUENCODE_MAX_READ 128*1024 /* in bytes */
```

Comment:
Bid lookahead is very different from the max line length.  Uuencode is designed to allow for arbitrary text at the beginning of the file, so the bid lookahead needs to allow for many lines.  We should probably have two constants here:
* UUENCODE_BID_MAX_READ of 128k as before
* UUENCODE_MAX_LINE_LENGTH which should be smaller.

A single line of _encoded data_ in a uuencoded file can be at most 64 bytes (62 bytes plus end-of-line).  The `begin` line can be longer than this, since it holds an arbitrary filename.  I would be happy to reduce the maximum line length to 32k or even smaller.

Reply:
I've adjusted the code to use 32k as suggested. We could reduce further, but as you mentioned, filename lengths can be arbitrary. We could speed up the runtime from O(n^2) to O(n) but I doubt that it's worth to reduce readability for such unlikely files.

Reply:
Much nicer!